File: LC1679.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        c = 0
        nums.sort()
        l = 0
        h = len(nums) - 1

        while l < h:
            t = nums[l] + nums[h]

            if t == k:
                c += 1
                h -= 1
                l += 1
            elif t > k:
                h -= 1
            else:
                l += 1
        
        return c

File: LC1679.cpp
class Solution {
public:
    int maxOperations(vector<int>& nums, int k) {
        sort(nums.begin(), nums.end());
        int l = 0, h = nums.size()-1, count = 0;

        while(l < h){
            int tar = nums[l] + nums[h];

            if(tar == k){
                count++;
                l++;
                h--;
            }
            else if(tar > k)
                h--;
            else
                l++;
        }

        return count;
    }
};
